### DISCUSSION TITLE
Navigation in Avalonia

### DISCUSSION THREAD

**Discussion by CosminCAS123** [AvaloniaUI/Avalonia#15470] [Q&A]
Hey , i got a question about navigation(for desktop). I am doing it by binding the content to a CurrentViewModel, and whenever I assign a new viewmodel, the view changes accordingly. But, isn't this very slow? If one of my views needs data from a database , whenever I create its viewmodel, it will have to call to the database just like it did the first time I accessed the view. Is there any way to cache that data? I am still new...and confused. Thanks!

**Comment by thevortexcloud** [comment]
You can just reuse the view model. You don't have to recreate it. For example you can just store your pages inside a dictionary and look up the view model when the user wants to switch pages.
